ERCS/erc-7583.md

```diff
@@ -0,0 +1,363 @@
+
+```solidity
+pragma solidity ^0.8.0;
+/**
+ * @title ERC-7583 token receiver interface 
+ * 
+ * Note: the ERC-165 identifier for this interface is 0xee1c3373.
+ */
+interface IERC7583Receiver {
+    /**
+    * @notice Handle the receipt of an inscription
+    * @dev The ERC-7583 smart contract calls this function on the recipient
+    *  after a `transfer`. This function MAY throw to revert and reject the
+    *  transfer. Return of other than the magic value MUST result in the
+    *  transaction being reverted.
+    *  Note: the contract address is always the message sender.
+    *  - _operator The address which called `safeTransferFrom` function
+    *  - _from The address which previously owned the token
+    *  - _tokenId The insctiption identifier which is being transferred
+    * @dev return `bytes4(keccak256("onERC7583Received(address,address,uint256)"))`
+    *  unless throwing
+    */
+    function onERC7583Received(address _operator, address _from, uint256 _tokenId) external returns(bytes4);
+}
+```
+
+The metadata extension is OPTIONAL for ERC-7583 smart contracts. This allows your smart contract to be interrogated for its name, symbol, decimals and for details about the assets which your inscriptions represent.
+
+```solidity
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Interface for the OPTIONAL metadata functions from the ERC-7583 standard.
+ *  Note: the ERC-165 identifier for this interface is 0xe45d67f9.
+ */
+interface IERC7583Metadata /* is IERC7583 */ {
+    /**
+     * @dev Embed the inscription data corresponding to the insId into Ethereum through events.
+     */
+    event Inscribe(uint256 indexed id, bytes data);
+    
+    /**
+     * @dev Returns the name of the inscription token.
+     */
+    function name() external view returns (string memory);
+
+    /**
+     * @dev Returns the symbol of the inscription token.
+     */
+    function symbol() external view returns (string memory);
+
+    /**
+     * @dev Returns the decimals places of the inscription token.
+     */
+    function decimals() external view returns (uint8);
+}
+```
+
+## Rationale
+
+The BRC20 protocol is a standard within the Bitcoin ecosystem that introduces a new way of transacting Fungible Tokens (FTs), allowing these tokens to be traded in a manner similar to Non-Fungible Tokens (NFTs). Specifically, the BRC20 protocol enables a number of FTs to be bundled together and traded as a single entity, offering more flexibility and possibilities for fungible token transactions. Within the Bitcoin ecosystem, the BRC20 protocol is primarily executed through off-chain interpreters. This approach relies on external participants and infrastructure to recognize and process these special transactions.
+
+In the Ethereum Virtual Machine (EVM), smart contracts provide powerful computational capabilities and a flexible programming model, making it feasible to implement the transaction modes described by the BRC20 protocol on Ethereum.
```

Comment:
```suggestion
```

None of this text is explaining technical decisions made within this proposal. You might be able to move this to Motivation if you'd like to keep it.

The Rationale section should be used to explain choices made within the EIP itself, while the Motivation section justifies the EIP as a whole. My favourite analogy is:

> Motivation: why do we need to build a shed?
> Rationale: why did we decide to paint the shed red?